calc/l3_route_resolver.go

```diff
@@ -783,7 +810,18 @@ func (r RouteInfo) IsValidRoute() bool {
 		r.Block.NodeName != "" ||
 		len(r.Host.NodeNames) > 0 ||
 		r.Pool.NATOutgoing ||
-		r.LocalWEP.RefCount > 0
+		len(r.WEP.RefCount) > 0
+}
+
+// Copy returns a copy of the RouteInfo. Since some fields are pointers, we need to
+// explicitly copy them so that they are not shared between the copies.
+func (r RouteInfo) Copy() RouteInfo {
+	cp := r
+	cp.WEP.RefCount = map[string]int{}
```

Comment:
For occupancy's sake, I think I'd handle the nil map case explicitly.  I.e. if we have 0 refcounts to store, set the map to nil and explicitly check/initialise it if we have something to store.

If you switch to using a slice, likewise; I'd set it to nil whenever it's length 0.  That also makes the Equals() method work in all cases; right now, it'll gte tripped up by the difference between a nil map and an empty map.

Reply:
Same goes for the zero check.  Don't want a 0-length slice to prevent cleanup